drivers/Timer.h

```diff
@@ -92,21 +97,27 @@ class Timer : private NonCopyable<Timer> {
 
     /** An operator shorthand for read()
      */
+    MBED_DEPRECATED_SINCE("mbed-os-6.0.0", "Floating point operators should normally be avoided for code size. If really needed, you can use `duration<float>(read_duration()).count()`")
     operator float();
 
     /** Get in a high resolution type the time passed in microseconds.
```

Comment:
From a design point of view, what's the purpose of this function? It's the same as `read_us` but the return type.

Reply:
Well, it's the strongly-typed Chrono return. You need it when using Chrono APIs. The question is should we deprecate `read_us`? See other comment.